sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConsumer.cs

```diff
@@ -182,6 +216,83 @@ protected EventHubConsumer()
+
+        /// <summary>
+        ///   Subscribes to events for the associated partition in the form of an asynchronous enumerable that sends events as they
+        ///   become available on the partition, waiting as necessary should there be no events available.
+        ///
+        ///   If the <paramref name="maximumWaitTime" /> is passed, if no events were available before the wait time elapsed, an empty
+        ///   item will be sent in the enumerable in order to return control and ensure that <c>await</c> calls do not block for an
+        ///   indeterminate length of time.
+        /// </summary>
+        ///
+        /// <param name="maximumWaitTime">The maximum amount of time to wait to for an event to be available before emitting an empty item; if <c>null</c>, empty items will not be published.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>An <see cref="IAsyncEnumerable{T}"/> to be used for iterating over events in the partition.</returns>
+        ///
+        /// <remarks>
+        ///   Unlike calls to <see cref="ReceiveAsync"/>, each subscriber to events is presented with an independent iterator; if there are multiple
+        ///   subscribers, they will each receive their own copy of an event to process, rather than competing for them.
+        ///
+        ///   Subscriptions will still compete with <see cref="ReceiveAsync" /> calls, however.  It is recommended that consumers either subscribe to
+        ///   events or explicitly receive batches, but do not use both approaches concurrently.
+        /// </remarks>
+        ///
+        /// <seealso cref="SubscribeToEvents(CancellationToken)"/>
+        ///
+        public virtual IAsyncEnumerable<EventData> SubscribeToEvents(TimeSpan? maximumWaitTime,
+                                                                     CancellationToken cancellationToken = default)
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return CreateEmptyAsyncEnumerable<EventData>();
+            }
+
+            try
+            {
+                var maximumQueuedEvents = Math.Min((Options.PrefetchCount / 4), (BackgroundPublishReceiveBatchSize * 2));
+                var subscription = SubscribeToChannel(EventHubPath, PartitionId, ConsumerGroup, maximumQueuedEvents, cancellationToken);
+
+                return new ChannelEnumerableSubscription<EventData>(subscription.ChannelReader, maximumWaitTime, () => UnsubscribeFromChannelAsync(subscription.Identifier), cancellationToken);
+            }
+            catch (Exception ex) when
+                (ex is TaskCanceledException
+                || ex is OperationCanceledException)
+            {
```

Comment:
Why suppress the exception?

Reply:
They're expected when cancellation is requested during semaphore acquisition or a `Task.Delay`; they aren't an actual exception case.   I don't believe that they should be surfaced; we should follow our normal clean up for when cancellation is triggered.